File: AdaptiveTD/AdaptiveTD/AdaptiveTD/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using System.Reflection;
using Microsoft.Xna.Framework;

namespace AdaptiveTD
{
    public enum EventType
    {
        [Description("build")]
        build,
        [Description("sell")]
        sell,
        [Description("upgrade")]
        upgrade
    }
    public class Event
    {
        EventType type;
        public EventType Type
        {
            get { return type; }
        }
        Vector2 tilePosition;
        public Vector2 TilePosition
        {
            get { return tilePosition; }
        }
        string towerType;

        public string TowerType
        {
            get { return towerType; }
        }

        public Event(EventType type, Vector2 tilePosition, string towerType)
        {
            this.type = type;
            this.tilePosition = tilePosition;
            this.towerType = towerType;
        }

        public string SaveString()
        {
            return "e:"+GetDescription(type) + ":" + towerType + ":" + tilePosition.X.ToString() + ":" + tilePosition.Y.ToString();
        }

        public static string GetDescription(object enumValue)
        {
            string defDesc = "";
            FieldInfo fi = enumValue.GetType().GetField(enumValue.ToString());

            if (null != fi)
            {
                object[] attrs = fi.GetCustomAttributes(typeof(DescriptionAttribute), true);
                if (attrs != null && attrs.Length > 0)
                    return ((DescriptionAttribute)attrs[0]).Description;
            }

            return defDesc;
        }
    }
}

File: LibGDXProjects/AdaptiveTD/src/no/uia/adaptiveTD/Event.java
package no.uia.adaptiveTD;

import com.badlogic.gdx.math.Vector2;

public class Event {

    EventType type;
    public EventType getType()
    {
        return type;
    }
    Vector2 tilePosition;
    public Vector2 getTilePosition()
    {
        return tilePosition;
    }
    String towerType;
    public String getTowerType()
    {
        return towerType;
    }

    public Event(EventType type, Vector2 tilePosition, String towerType)
    {
        this.type = type;
        this.tilePosition = tilePosition;
        this.towerType = towerType;
    }

    public String SaveString()
    {
        return "e:"+ type.name() + ":" + towerType + ":" + tilePosition.x + ":" + tilePosition.y;
    }
}
